### DISCUSSION TITLE
Using PlusServer with an eye-tracker

### DISCUSSION THREAD

**Discussion by DylanM-ihu** [PlusToolkit/PlusLib#1057] [Q&A]
Hello,

I am working with an eye-tracker, the GP3HD from GazePoint, and I would like to collect data with PlusServer (to synchronize the eye-tracker data with US images and US probe poses). The eye-tracker API (Open Gaze API) uses a standard TCP/IP socket for communication (XML data format) between an application and the eye-tracker.

Do you know if PlusServer has already been used with an eye-tracker? Or do you have any advice for me to develop my own solution using the eye-tracker API?

Thanks a lot!

**Comment by Sunderlandkyl** [comment]
I experimented with the Tobii eye tracker (see branch [here](https://github.com/Sunderlandkyl/PlusLib/tree/tobii_eye_tracking/src/PlusDataCollection/TobiiInteractionSDK)). It worked well, although we didn't end up using it due to the restrictive nature of the license.

It's always good to start by using another device as an example. You can find some additional documentation on the process of adding a new device here: https://plustoolkit.github.io/devicecode.

**Reply by DylanM-ihu** [reply]
Thank you for your answer. I read the ressources you provided me but I am not sure that it applies to my situation since I do not have SDK/API include directories and library files. To communicate with the eye-tracker I have to open an application called GazePoint Control that will collect the data and then the API allows me to send XML command to GazePoint Control via a standard TCP/IP socket.

Do you know how I can connect Plus to this TCP/IP interface and send command (using 3D Slicer?)?

I simulated a virtual serial port and connected it to the TCP/IP interface (e.g. COM2 -> 127.0.0.1, on port 4242 for my eye-tracker), this way using a GenericSerialDevice, I was able to connect Plus to the TCP/IP interface but I do not know how to send XML command.
